lib/stat_generator.rb

```diff
@@ -149,6 +149,18 @@ def total_games_played_by_team(team_id)
     end
   end
 
+  def total_games_played_by_team_when_away(team_id)
+    @game_teams.count do |game_team|
+      game_team.team_id == team_id && game_team.home_or_away == "away"
+    end
+  end
+
+  def total_games_played_by_team_when_home(team_id)
+    @game_teams.count do |game_team|
+      game_team.team_id == team_id && game_team.home_or_away == "home"
+    end
+  end
+
```

Comment:
I think these two methods can be combined into one. Just replace "home" or "away" with a variable passed in through the argument